identity-account/src/updates/update.rs

```diff
@@ -0,0 +1,465 @@
+        ensure!(
+          state.document().service().query(&service_url).is_some(),
+          UpdateError::ServiceNotFound
+        );
+
+        state.document_mut().remove_service(service_url)?;
+      }
+    }
+
+    state.document_mut().set_updated(Timestamp::now_utc());
+
+    Ok(())
+  }
+}
+
+async fn insert_method_secret(
+  store: &dyn Storage,
+  did: &IotaDID,
+  location: &KeyLocation,
+  method_type: MethodType,
+  method_secret: MethodSecret,
+) -> Result<PublicKey> {
+  match method_secret {
+    MethodSecret::Ed25519(private_key) => {
+      ensure!(
+        private_key.as_ref().len() == ed25519::SECRET_KEY_LENGTH,
+        UpdateError::InvalidMethodSecret(format!(
+          "an ed25519 private key requires {} bytes, found {}",
+          ed25519::SECRET_KEY_LENGTH,
+          private_key.as_ref().len()
+        ))
+      );
+
+      ensure!(
+        matches!(method_type, MethodType::Ed25519VerificationKey2018),
+        UpdateError::InvalidMethodSecret(
+          "MethodType::Ed25519VerificationKey2018 can only be used with an ed25519 method secret".to_owned(),
+        )
+      );
+
+      store.key_insert(did, location, private_key).await
+    }
+    MethodSecret::MerkleKeyCollection(_) => {
+      ensure!(
+        matches!(method_type, MethodType::MerkleKeyCollection2021),
+        UpdateError::InvalidMethodSecret(
+          "MethodType::MerkleKeyCollection2021 can only be used with a MerkleKeyCollection method secret".to_owned(),
+        )
+      );
+
+      todo!("[Update::CreateMethod] Handle MerkleKeyCollection")
+    }
+  }
+}
+
+// =============================================================================
+// Update Builders
+// =============================================================================
+
+impl_update_builder!(
+/// Create a new method on an identity.
+///
+/// # Parameters
+/// - `type_`: the type of the method, defaults to [`MethodType::Ed25519VerificationKey2018`].
+/// - `scope`: the scope of the method, defaults to [`MethodScope::default`].
+/// - `fragment`: the identifier of the method in the document, required.
+/// - `method_secret`: the secret key to use for the method, optional. Will be generated when omitted.
+CreateMethod {
+  @defaulte type_ MethodType = Ed25519VerificationKey2018,
+  @default scope MethodScope,
+  @required fragment String,
+  @optional method_secret MethodSecret
+});
+
+impl_update_builder!(
+/// Delete a method on an identity.
+///
+/// # Parameters
+/// - `fragment`: the identifier of the method in the document, required.
+DeleteMethod {
+  @required fragment String,
+});
+
+impl_update_builder!(
+/// Attach one or more verification relationships to a method on an identity.
+///
+/// # Parameters
+/// - `relationships`: the relationships to add, defaults to an empty [`Vec`].
+/// - `fragment`: the identifier of the method in the document, required.
+AttachMethod {
+  @required fragment String,
+  @default relationships Vec<MethodRelationship>,
+});
```

Comment:
Discussion: should `AttachMethod` and `DetachMethod` take in a single relationship instead of a `Vec`? I know the PR didn't change this, just wondering since creating a `Vec` isn't very ergonomic, even more so in bindings.

Reply:
It would make error handling easier, if we want to return an error if a relationship wasn't found/already existed. If 4 relationships were added successfully, but the fifth one was already there, then what do you return? On the other hand, this is an argument to not return an error for these methods, because they're idempotent.

Also, the ergonomics are quite nice, since you don't have to pass a vector but can just call `relationship` multiple times to add multiple relationships, e.g.:

```rust
  account
    .update_identity()
    .attach_method()
    .fragment("my-next-key")
    .relationship(MethodRelationship::CapabilityDelegation)
    .relationship(MethodRelationship::CapabilityInvocation)
    .apply()
    .await?;
```

As long as the builder works nicely in other languages, that should translate nicely. If that's not idiomatic, then for example in JS or Python, passing a list seems nice, too. E.g.:

```python
await account.attach_method("my-next-key", [MethodRelationship.CapabilityDelegation, MethodRelationShip.CapabilityInvocation])
```

Overall, I think it's fine.

Reply:
> Also, the ergonomics are quite nice, since you don't have to pass a vector but can just call relationship multiple times to add multiple relationships, e.g.:

Good point, I didn't catch that!

> On the other hand, this is an argument to not return an error for these methods, because they're idempotent.

This change would be for if those methods were not idempotent, as long as they are (and given the builder ergonomics) this is absolutely fine. Thank you.